### ISSUE TITLE
Incremental compilation does not catch type errors in deep imports

### ISSUE THREAD

**Issue by horup** [Microsoft/TypeScript#24986]
It seems watch mode does not check for type errors induced by changes to types in deep imports.
A restart is required to catch these errors.

***TypeScript Version:**** 
Version 3.0.0-dev.20180609 on Windows 10

**Search Terms:** 
watch, incremental, deep

**Code**
Three files> a.ts, b.ts and c.ts + empty tsconfig.json file to please tsc.
```ts
// a.ts:
import {B} from './b'; 
let b = new B(); 
console.log(b.c.d);

// b.ts:
import {C} from './c';
export class B
{
    c = new C();
}

// c.ts:
export class C
{
    d = 1;
}

```

**Expected behavior:**
Running tsc --watch and making a type error in a.ts by inducing a change in c.ts should result in a type error in the console.

e.g. changing 
```ts
    d = 1;
```
to 
```ts
    d2 = 1;
```
should result in a type error in a.ts:
a.ts:3:17 - error TS2339: Property 'd' does not exist on type 'C'.

**Actual behavior:**
Running tsc --watch and making a type error in a.ts by inducing a change in c.ts does not result in a type error. Instead, the compiler happily writes:
Found 0 errors. Watching for file changes.

Restarting tsc in watch mode results in the expected type error:
a.ts:3:17 - error TS2339: Property 'd' does not exist on type 'C'.

**Comment by mhegazy** [comment]
looks like a duplicate of https://github.com/Microsoft/TypeScript/issues/25068

**Comment by sheetalkamat** [comment]
@mhegazy this does not seem like duplicate of #25068. The issue here is when c.ts is changed the declaration emit for c.ts changes which results in emitting b.ts but since b.ts does not change, a.ts is not emitted.
